tools/code-analyzer/src/git.ts

```diff
@@ -5,12 +5,135 @@ import { CliUx } from '@oclif/core';
 import { execSync } from 'child_process';
 import { join } from 'path';
 import { tmpdir } from 'os';
-import { readFileSync } from 'fs';
+import { readFileSync, rmSync } from 'fs';
+import { simpleGit } from 'simple-git';
+import { v4 } from 'uuid';
 
 /**
  * Internal dependencies
  */
 import { startWPEnv, stopWPEnv, isValidCommitHash } from './utils';
+/**
+ * Clone a git repository.
+ *
+ * @param {string} repoPath- the path (either URL or file path) to the repo to clone.
+ * @param          repoPath
+ * @return {string} the path to the cloned repo.
+ */
+export const cloneRepo = async ( repoPath: string ) => {
+	const tempFolderName = v4();
+	const git = simpleGit( { baseDir: tmpdir() } );
+	await git.clone( repoPath, tempFolderName );
+
+	return tempFolderName;
+};
+
+/**
+ * Do a git diff of 2 commit hashes (or branches)
+ *
+ * @param {string} baseDir - baseDir that the repo is in
+ * @param {string} hashA   - either a git commit hash or a git branch
+ * @param {string} hashB   - either a git commit hash or a git branch
+ * @return {string} - diff of the changfiles between the 2 hashes
+ */
+export const diffHashes = ( baseDir: string, hashA: string, hashB: string ) => {
+	const git = simpleGit( { baseDir } );
+	return git.diff( [ `${ hashA }..${ hashB }` ] );
+};
+
+/**
+ * Determines if a string is a commit hash or not.
+ *
+ * @param {string} ref - the ref to check
+ * @return {boolean} whether the ref is a commit hash or not.
+ */
+const refIsHash = ( ref: string ) => {
```

Comment:
Previously I used `isValidCommitHash` function to perform this task. It uses `git show -s` to not only check if its a hash, but exists in the history. This seems like a good opportunity for an informative error message if a hash isn't in the history.

Reply:
Yeah fair call that would be a better and more robust check. I'll fix that.

Reply:
So I still want to sometimes check if something is a hash or not, but when trying to get the commit hashes for each ref I now use your `git show` approach to bail if they're not valid. https://github.com/woocommerce/woocommerce/pull/33837/commits/5ea9a62686c88e6d768ab2c6c546983a369bdbbe